frontend/components/utils/BidInput.vue

```diff
@@ -0,0 +1,136 @@
+                    @focus="hideTotalPrice()"
+                    @blur="showTotalPriceIfEmpty()"
+                />
+            </label>
+        </Tooltip>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import { Tooltip } from 'ant-design-vue';
+import FormatCurrency from './FormatCurrency.vue';
+
+export default Vue.extend({
+    components: { FormatCurrency, Tooltip },
+    props: {
+        totalPrice: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        minimumDepositDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            required: true,
+        },
+        amountToBid: {
+            type: Object as Vue.PropType<BigNumber> | undefined,
+            default: undefined,
+        },
+        disabled: {
+            type: Boolean,
+            default: false,
+        },
+    },
+    data() {
+        return {
+            amountToBidInput: '' as string,
+        };
+    },
+    computed: {
+        amountToBidInputParsed(): BigNumber | undefined {
+            if (!this.amountToBidInput) {
+                return undefined;
+            }
+            const amountToBid = new BigNumber(this.amountToBidInput);
+            return amountToBid;
+        },
+        error(): string | undefined {
+            if (!this.amountToBidInputParsed) {
+                return undefined;
+            }
+            const maximumBid = this.totalPrice.minus(this.minimumDepositDai);
+            let error = undefined as string | undefined;
+            if (this.amountToBidInputParsed.isGreaterThan(maximumBid)) {
+                error = `The bidding amount can not be greater than ${maximumBid.toFixed(2)} DAI`;
+            }
+            if (this.amountToBidInputParsed.isLessThan(this.minimumDepositDai)) {
+                error = `The bidding amount can not be smaller than ${this.minimumDepositDai.toFixed(2)} DAI`;
+            }
+            return error;
+        },
+    },
+    watch: {
+        amountToBidInputParsed(newVal, oldVal) {
+            if (this.error) {
+                this.$emit('update:amountToBid', new BigNumber(NaN));
+                return;
+            }
+            if (newVal && newVal.isNaN()) {
+                this.amountToBidInput = oldVal;
+                return;
+            }
+            if (newVal) {
+                this.$emit('update:amountToBid', newVal);
+            }
+        },
+        amountToBid(newVal: BigNumber | undefined) {
+            if (!newVal) {
+                this.amountToBidInput = '';
+                return;
+            }
+            if (newVal.isEqualTo(this.minimumDepositDai)) {
+                this.amountToBidInput = newVal.toString();
+            }
+        },
+    },
+    methods: {
+        hideTotalPrice(): void {
+            if (!this.amountToBid) {
+                this.$emit('update:amountToBid', new BigNumber(0));
```

Comment:
0 is actually an invalid input, since it's below minimum, therefore, I would better set it to NaN
```suggestion
                this.$emit('update:amountToBid', new BigNumber(NaN));
```